ofrak_core/ofrak/core/patch_maker/modifiers.py

```diff
@@ -53,41 +101,23 @@ class PatchFromSourceModifier(Modifier):
 
     targets = (Program,)
 
-    async def _flush_source_bundle_to_disk(
-        self, resource_id: bytes, prototype_resource: Resource
-    ) -> str:
-        path = tempfile.mkdtemp()
-        source_dir_resource = await self._resource_factory.create(
-            prototype_resource.get_job_id(),
-            resource_id,
-            prototype_resource.get_resource_context(),
-            prototype_resource.get_resource_view_context(),
-            prototype_resource.get_component_context(),
-            prototype_resource.get_job_context(),
-        )
-        source_bundle = await source_dir_resource.view_as(SourceBundle)
-        await source_bundle.flush_to_disk(path)
-        return path
-
     async def modify(self, resource: Resource, config: PatchFromSourceModifierConfig) -> None:
 
         if config.patch_name is None:
-            patch_name = f"{config.source_code_resource_id.hex()}_{resource.get_id().hex()}"
+            patch_name = f"{resource.get_id().hex()}_patch"
         else:
             patch_name = config.patch_name
 
         build_tmp_dir = tempfile.mkdtemp()
 
-        source_tmp_dir = await self._flush_source_bundle_to_disk(
-            config.source_code_resource_id, resource
-        )
+        source_tmp_dir = tempfile.mkdtemp()
+        config.source_code.dump(source_tmp_dir)
 
         header_dirs = []
-        if config.header_directory_resource_ids:
-            for header_directory_id in config.header_directory_resource_ids:
-                header_dirs.append(
-                    await self._flush_source_bundle_to_disk(header_directory_id, resource)
-                )
+        for header_directory in config.header_directories:
+            header_tmp_dir = tempfile.mkdtemp()
+            header_directory.dump(header_tmp_dir)
+            header_dirs.append(header_tmp_dir)
```

Comment:
Unless I'm misunderstanding something, the call to `tempfile.mkdtemp` leaks temporary directories, right? The [docs for `mkdtemp`](https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp) specify that the caller is responsible for cleaning up, but I don't see logic to do that anywhere.

Reply:
I don't think we do want to clean it up by deleting it. I often find it useful for debugging to re-run the commands that PatchMaker invoked, which operate on files in those temp directories. Even when the component succeeds, it's useful to still have the files there if I want to inspect them or try modifying the commands for a similar result. 

Over time it leaves old folders in /tmp, but that's kind of what /tmp is for.